rancher2/resource_rancher2_cluster.go

```diff
@@ -531,14 +531,19 @@ func findFlattenClusterRegistrationToken(client *managementClient.Client, cluste
 func findClusterRegistrationToken(client *managementClient.Client, clusterID string) (*managementClient.ClusterRegistrationToken, error) {
 	for i := range clusterRegistrationTokenNames {
 		regTokenID := clusterID + ":" + clusterRegistrationTokenNames[i]
-		regToken, err := client.ClusterRegistrationToken.ByID(regTokenID)
-		if err != nil {
-			if !IsNotFound(err) {
-				return nil, err
+		for retry, retries := 1, 10; retry <= retries; retry++ {
+			regToken, err := client.ClusterRegistrationToken.ByID(regTokenID)
+			if err != nil {
+				if !IsNotFound(err) {
+					return nil, err
+				}
+				break
+			}
+			if (len(regToken.Command) > 0 && len(regToken.NodeCommand) > 0) || retry == retries {
+				return regToken, nil
```

Comment:
Isn't reaching maximum retries and not having a regToken considered an error? Or is this expected and handled by when its called? If it's empty and retries has been reached, regToken is not complete and can't be used so it should be an error or no?

Reply:
It's indeed.  If the regToken is found and retries has been reached, a partial regToken is returned instead of error. The regToken is just a computed field for the cluster resource.